lib/shopt.sh

```diff
@@ -19,18 +19,26 @@ PROMPT_DIRTRIM=2
 bind Space:magic-space
 
 # Turn on recursive globbing (enables ** to recurse all directories)
-shopt -s globstar 2> /dev/null                                                                                   
+shopt -s globstar 2> /dev/null
 
-# Case-insensitive globbing (used in pathname expansion)
-shopt -s nocaseglob;
+# Case-insensitive globbing (used in pathname expansion) and matching
+# (used in case, [[]], word expansions and command completions)
+if [[ ${OMB_CASE_SENSITIVE:-${CASE_SENSITIVE:-}} == true ]]; then
+   shopt -u nocaseglob
+   shopt -u nocasematch
+else
+   shopt -s nocaseglob
+   shopt -s nocasematch
+fi
 
 ## SMARTER TAB-COMPLETION (Readline bindings) ##
 
-# Perform file completion in a case insensitive fashion
+# Conditionally perform file completion in a case insensitive fashion.
+# OMB_CASE_SENSITIVE to 'true' will switch from the default, case
+# insensitive, matching.
```

Comment:
```suggestion
# insensitive, matching to the case-sensitive one.
```
It is confusing to only mention the behavior before applying the setting.

Reply:
> It is confusing to only mention the behavior before applying the setting.

Sure, that makes sense to me.